test/unittests/libp2p/net.cpp

```diff
@@ -939,6 +940,85 @@ BOOST_AUTO_TEST_CASE(findNodeIsSentAfterPong)
     BOOST_CHECK_EQUAL(datagram3->typeName(), "FindNode");
 }
 
+BOOST_AUTO_TEST_CASE(pingNotSentAfterPongForKnownNode)
+{
+    // Validate 3 scenarios:
+    // * A ping is sent after a pong for a new node
+    // * No ping is sent after a pong for a known node with a valid endpoint proof
+    // * A ping is sent after a pong for a known node with an invalid endpoint proof
+
+    TestNodeTableHost nodeTableHost1(0);
+    nodeTableHost1.start();
+    auto& nodeTable1 = nodeTableHost1.nodeTable;
+
+    TestUDPSocketHost nodeSocketHost2{getRandomPortNumber()};
+    nodeSocketHost2.start();
+    auto nodePort2 = nodeSocketHost2.port;
+    auto nodeEndpoint2 =
+        NodeIPEndpoint{bi::address::from_string(c_localhostIp), nodePort2, nodePort2};
+
+    // Add node to the node table to trigger ping
+    auto nodeKeyPair2 = KeyPair::create();
+    Node node2(nodeKeyPair2.pub(), nodeEndpoint2);
+    nodeTable1->addNode(node2);
+
+    // Verify ping is received
+    auto packetReceived1 = nodeSocketHost2.packetsReceived.pop(chrono::milliseconds(5000));
+    auto datagram1 =
+        DiscoveryDatagram::interpretUDP(bi::udp::endpoint{}, dev::ref(packetReceived1));
+    auto ping1 = dynamic_cast<PingNode const&>(*datagram1);
+    BOOST_CHECK_EQUAL(datagram1->typeName(), "Ping");
```

Comment:
`dynamic_cast` above would throw anyway if it's not Ping

Reply:
Ah the line before the BOOST_CHECK_EQUAL is a copy/paste artifact. Thanks for catching this!